convert/src/main/kotlin/com/github/zeldigas/text2confl/convert/confluence/ReferenceProvider.kt

```diff
@@ -39,24 +44,41 @@ class ReferenceProviderImpl(private val basePath: Path, documents: Map<Path, Pag
     ReferenceProvider {
 
     companion object {
-        private val URI_DETECTOR = "^[a-zA-Z][a-zA-Z0-9.+-]+:/{0,2}".toRegex()
+        private const val URI_DETECTOR = "^(https?|ftp)://[-a-zA-Z0-9+&@#/%?=~_|!:,.;'()]*[-a-zA-Z0-9+&@#/%=~_|]"
+        private const val MAILTO_DETECTOR = "mailto:"
+        private const val LOCALHOST_DETECTOR = "localhost:"
+
+        private val logger = KotlinLogging.logger {}
+    }
+    fun isValid(url: String): Boolean {
+        val pattern = Pattern.compile(URI_DETECTOR, Pattern.CASE_INSENSITIVE);
+        val matcher = pattern.matcher(url.trim());
+        return matcher.matches();
     }
 
     private val normalizedDocs =
         documents.map { (path, header) -> path.relativeTo(basePath).normalize() to header }.toMap()
 
     override fun resolveReference(source: Path, refTo: String): Reference? {
-        if (URI_DETECTOR.matches(refTo)) return null
+
+        if (refTo.startsWith(MAILTO_DETECTOR)) return null
+        if (refTo.startsWith(LOCALHOST_DETECTOR)) return null
+        if (isValid(refTo)) return null
         if (refTo.startsWith("#")) return Anchor(refTo.substring(1))
 
         val parts = refTo.split("#", limit = 2)
         val ref = parts[0]
         val anchor = parts.getOrNull(1)
 
-        val targetPath = source.resolveSibling(ref).relativeTo(basePath).normalize()
+        try {
 
-        val document = normalizedDocs[targetPath]?.title ?: return null
-        return Xref(document, anchor)
+            val targetPath = source.resolveSibling(ref).relativeTo(basePath).normalize()
+            val document = normalizedDocs[targetPath]?.title ?: return null
+            return Xref(document, anchor)
+        } catch (ex: InvalidPathException) {
+            logger.error { "Failed to resolve : $refTo  from $source" }
```

Comment:
it's a bad pattern - log error and rethrow exception - because you'd get one more logging in another place. Either do not log or throw another exception with chained root cause.